File: Jhanavi/12-09-2022/Intersection_of_Two_Linked_Lists.py
class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        ptr_a = headA
        ptr_b = headB
        
        while ptr_a != ptr_b:
            ptr_a = headB if not ptr_a else ptr_a.next
            ptr_b = headA if not ptr_b else ptr_b.next
            
        return ptr_a

File: Farhaan/Week-2 Problems/intersection-of-two-linked-lists.cpp
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB){
    ListNode *on=headA;
    ListNode *tn=headB;
    int c=0,count=0;
    while(on!=NULL){
        c++;
        on=on->next;
    }
    while(tn!=NULL){
        count++;
        tn=tn->next;
    }
    on=headA;
    tn=headB;
    if(c>count){
        for(int i=0;i<c-count;i++)
        on=on->next;
    }
    else{
        for(int i=0;i<count-c;i++)
        tn=tn->next;
    }
    while(tn!=on){
        if(tn==NULL)
            return NULL;
        if(on==NULL)
            return NULL;
        tn=tn->next;
        on=on->next;
    }
    return on;
    }
};